tests/unit/meta/meta.ts

```diff
@@ -0,0 +1,201 @@
+import global from '@dojo/core/global';
+import Map from '@dojo/shim/Map';
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import { v } from '../../../src/d';
+import { ProjectorMixin } from '../../../src/main';
+import { Base as MetaBase } from '../../../src/meta/Base';
+import { WidgetBase } from '../../../src/WidgetBase';
+import { stub } from 'sinon';
+
+let rAF: any;
+
+function resolveRAF() {
+	for (let i = 0; i < rAF.callCount; i++) {
+		rAF.getCall(i).args[0]();
+	}
+	rAF.reset();
+}
+
+registerSuite({
+	name: 'meta base',
+
+	beforeEach() {
+		rAF = stub(global, 'requestAnimationFrame').returns(1);
+	},
+
+	afterEach() {
+		rAF.restore();
+	},
+
+	'meta returns a singleton'() {
+		class TestMeta extends MetaBase {
+		}
+
+		class TestWidget extends ProjectorMixin(WidgetBase)<any> {
+			render() {
+				return v('div', {
+					innerHTML: 'hello world',
+					key: 'root'
+				});
+			}
+
+			getMeta() {
+				return this.meta(TestMeta);
+			}
+		}
+
+		const div = document.createElement('div');
+
+		const widget = new TestWidget();
+		widget.append(div);
+
+		assert.strictEqual(widget.getMeta(), widget.getMeta());
+	},
+
+	'meta is provided a list of nodes with keys'() {
+		class TestMeta extends MetaBase {
+		}
+
+		class TestWidget extends ProjectorMixin(WidgetBase)<any> {
+			render() {
+				return v('div', {
+					innerHTML: 'hello world',
+					key: 'root'
+				});
+			}
+
+			getMeta() {
+				return this.meta(TestMeta);
+			}
+		}
+
+		const div = document.createElement('div');
+
+		const widget = new TestWidget();
+		widget.append(div);
+		const meta = widget.getMeta();
+
+		assert.isTrue((<any> meta).nodes.size > 0);
```

Comment:
It's a protected instance variable. Should I just cast it to `{ nodes: Map<string, HTMLElement> }`?